site/app/controllers/course/CourseRegistrationController.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace app\controllers\course;
+
+use app\controllers\AbstractController;
+use app\controllers\admin\ConfigurationController;
+use app\libraries\response\RedirectResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use app\models\Email;
+
+class CourseRegistrationController extends AbstractController {
+    /**
+     * @param array<string> $instructors
+     */
+    private function notifyInstructors(string $user, string $term, string $course, array $instructors): void {
+        $subject = "Self-registration of $user for course $course";
+        $body = "Student $user has self-registered for course $course for term $term.";
+        $emails = [];
+        foreach ($instructors as $instructor) {
+            $emails[] = new Email(
+                $this->core,
+                [
+                    "subject" => $subject,
+                    "body" => $body,
+                    "to_user_id" => $instructor
+                ]
+            );
+        }
+
+        $this->core->getNotificationFactory()->sendEmails($emails);
+    }
+
+    #[Route("/courses/{term}/{course}/self_register")]
```

Comment:
Would it make sense to just make the route be `.../register`?  The "self" part seems implied.

Reply:
This has been updated to only be 'register'